docs/content/src/learn/guides/integration-patterns/content-based-routing/README.md

```diff
@@ -0,0 +1,99 @@
+
+## What you'll build
+
+We shall have a service ‘stockQuote’ which accepts an http request from a client. The client appends as a query parameter in the request, the company name of which he wants to know the price of the stock. The stockQuote service identifies the company, routes the request to the relevant company’s stock quote service, obtain the response from the company’s service and returns the response to the client. If the company name is not available in the request, the service will simply respond with a 400 - Bad Request.
+
+![cbr](../../assets/img/cbr.png)
+
+## Prerequisites
+
+- [Ballerina Distribution](https://ballerina.io/learn/getting-started/)
+- A Text Editor or an IDE 
+> **Tip**: For a better development experience, install the Ballerina IDE plugin for [VS Code](https://marketplace.visualstudio.com/items?itemName=ballerina.ballerina)
+	
+## Implementation
+
+* Create a new ballerina project named ‘content-based-routing’.
+
+```bash
+$ ballerina new content-based-routing
+```
+
+* Navigate to the directory content-based-routing.
+
+* Add a new module named ‘stockquote_service’ to the project.
+
+```bash
+$ ballerina add stockquote_service
+```
+
+* Open the project with VSCode. The project structure will be like the following.
+
+```shell
+.
+├── Ballerina.toml
+└── src
+    └── stockquote_service
+        ├── main.bal
+        ├── Module.md
+        ├── resources
+        └── tests
+            ├── main_test.bal
+            └── resources
+```
+
+We can remove the file `main_test.bal` at the moment, since we're not writing any tests for our service.
+
+First let's create the services that we shall be using as backend endpoints. 
+
+* Create a new file named `ibmService.bal` file under 'stockquote_service' with the following content.
+
+**ibmService.bal**
+<!-- INCLUDE_CODE: src/stockquote_service/ibmService.bal -->
+
+This is simply a service that will run on port 8081 responding a text payload `10000.00`.
+
+* Likewise, let's create another file `msService.bal` with the following content.
+
+**msService.bal**
+<!-- INCLUDE_CODE: src/stockquote_service/msService.bal -->
+
+* Now let's open the main.bal file and add the following content. This is going to be our integration logic.
+
+**main.bal**
+<!-- INCLUDE_CODE: src/stockquote_service/main.bal -->
+
+Here we’re calling two services we created earlier, using the endpoints ‘ibmEP’ and ‘msEP’.
+
+In the stockQuote service, the ‘company’ is retrieved as a query parameter. Then the value of the ‘company’ is checked. If it is ‘ibm’, the ‘ibmEP’ is called and its response is saved. If it is ‘ms’, the ‘msEP’ is called. If there is no value set, we’re simply setting a 400-Bad Request response. Finally the response is sent back to the client.
```

Comment:
```suggestion
In the stockQuote service, the `company` is retrieved as a query parameter. Then the value of the `company` is checked. If it is `ibm`, the `ibmEP` is called and its response is saved. If it is `ms`, the `msEP` is called. If there is no value set, we’re simply setting a 400-Bad Request response. Finally the response is sent back to the client.
```

Better if we can use a fictitious company instead of IBM. foo or something like that is fine.